index.js

```diff
@@ -24,33 +24,47 @@ module.exports = {
     return this.appConfig.podModulePrefix ? this.appConfig.podModulePrefix.replace(this.appConfig.modulePrefix, '') : '';
   },
 
+  _namespacingIsEnabled: function() {
+    return this.addonConfig.namespacing !== false;
+  },
+
   included: function(app) {
     this._super.included.apply(this, arguments);
     this.appConfig = app.project.config();
+    this.addonConfig = this.app.project.config(app.env)['ember-component-css'] || {};
     this.allowedStyleExtensions = app.registry.extensionsForType('css').filter(Boolean);
   },
 
   treeForAddon: function(tree) {
-    var podStyles = this._getPodStyleFunnel();
-    var podNames = new ExtractNames(podStyles, {
-      annotation: 'Walk (ember-component-css extract class names from style paths)'
-    });
+    if (this._namespacingIsEnabled()) {
+      var podStyles = this._getPodStyleFunnel();
+      var podNames = new ExtractNames(podStyles, {
+        annotation: 'Walk (ember-component-css extract class names from style paths)'
+      });
 
-    var treeAndNames = new Merge([tree, podNames], {
-      overwrite: true,
-      annotation: 'Merge (ember-component-css merge names with addon tree)'
-    });
+      var treeAndNames = new Merge([tree, podNames], {
```

Comment:
could tree just be redefined here so that way there is only one return statement?